freqtrade/exchange/exchange.py

```diff
@@ -69,6 +69,8 @@ class Exchange:
     }
     _ft_has: Dict = {}
 
+    _leverage_brackets: Dict
```

Comment:
Let me know what you think about this variable, if it should be renamed or implemeted differently

Reply:
for the moment, it's unclear to me what this variable tries to accomplish ...

Reply:
For kraken it just stores the leverage amounts allowed, but binances leverage is kind of different, you can see the structure of each by running the code below.

Binance returns an object where the key value pairs look like

```
 'ZIL/USDT': [[0.0, '0.01'],
              [5000.0, '0.025'],
              [25000.0, '0.05'],
              [100000.0, '0.1'],
              [250000.0, '0.125'],
              [1000000.0, '0.5']],
```

In this example, if you are trading up to 4999.99 ZIL, then you have 100x leverage max, if you are trading between 5000 and 24999.99 ZIL, then you have 25x leverage max, and so on?

the first number is the upper limit for the bracket to apply (in margin currency) and the second is the percentage of initial margin rate which is the same as inverted leverage

[Binance has a FAQ page that explains it](https://www.binance.com/en/support/faq/360033162192)

-----------------

You can run the code below with your own api key details to see the objects

```
binance_usdm = ccxt.binanceusdm({
    'api_key': config["binance"].api_key,
    'secret': config["binance"].secret,
    'options': {
        'defaultType': 'future'
    },
})

kraken = ccxt.kraken({
    'api_key': config["kraken"].api_key,
    'secret': config["kraken"].secret,
})

def lev_brackets(exchange: str):
    if exchange == "binance" or exchange == "binance_usdm":
        return binance_usdm.load_leverage_brackets()
    elif exchange == "kraken":
        leverages = {}
        for pair, market in kraken.load_markets().items():
            info = market['info']
            leverage_buy = info['leverage_buy']
            leverage_sell = info['leverage_sell']
            if len(info['leverage_buy']) > 0 or len(info['leverage_sell']) > 0:
                leverages[pair] = leverage_buy
        return leverages
        
pprint(lev_brackets("binance"))
#pprint(lev_brackets("kraken"))


```

Reply:
Still unclear? Or do you get it

Reply:
i guess it's clear now ... the more money ~you have~ you put on the line, the more leverage you can use.

I think your sample has it inverted though - according to the page, you get more leverage with lower trades (50x < 50.000), and then lowering the higher you go.

Reply:
I'm pretty sure the inversion is correct, they're not trying to reward you for trading more. The purpose of this is that on margin mode, they only have a certain amount of crypto to loan out, and on futures I'm not sure what the purpose is, maybe just a safeguard in case you get liquidated, but the price drops so fast that you end up with negative money and decide not to pay binance the difference

Reply:
The second number is the ratio of collateral to the notional value of the trade

Reply:
ah, in my understanding, the 2nd value was the maximum leverage that can be applied ... which did seem strange to me.

Reply:
is this resolved?

Reply:
i do have understood it if that's what you mean.

Naming of a private variable is ... not 100% important - but i'm not sure if "brakets" is appropriate, or if it's really "steps" (but again - variable names can easily be changed - and if "brakets" is an official term we should leave it at that)

Reply:
I called them brackets because that's what ccxt calls them, but it's not really the same thing for kraken

Reply:
leave it as is for now, then.